File: src/Java/CreationalPatterns/AbstractFactory/Furniture/src/ConcreteFactories/ContemporaryFurnitureFactory.java
package ConcreteFactories;

import Interfaces.IFurniture;
import Interfaces.IFurnitureFactory;
import Products.ContemporaryCabinet;
import Products.ContemporaryChair;
import Products.ContemporaryDiningTable;

public class ContemporaryFurnitureFactory implements IFurnitureFactory {
    @Override
    public IFurniture createCabinet() {
        return new ContemporaryCabinet();
    }

    @Override
    public IFurniture createChair() {
        return new ContemporaryChair();
    }

    @Override
    public IFurniture createDiningTable() {
        return new ContemporaryDiningTable();
    }
}

File: src/PHP/CreationalPatterns/AbstractFactory/Furniture/ConcreteFactories/ContemporaryFurnitureFactory.php
<?php

namespace CreationalPatterns\AbstractFactory\Furniture\ConcreteFactories;

use CreationalPatterns\AbstractFactory\Furniture\Interfaces\IFurniture;
use CreationalPatterns\AbstractFactory\Furniture\Interfaces\IFurnitureFactory;
use CreationalPatterns\AbstractFactory\Furniture\Products\ContemporaryCabinet;
use CreationalPatterns\AbstractFactory\Furniture\Products\ContemporaryChair;
use CreationalPatterns\AbstractFactory\Furniture\Products\ContemporaryDiningTable;

class ContemporaryFurnitureFactory implements IFurnitureFactory {
    public function createCabinet(): IFurniture {
        return new ContemporaryCabinet();
    }

    public function createChair(): IFurniture
    {
        return new ContemporaryChair();
    }

    public function createDiningTable(): IFurniture
    {
        return new ContemporaryDiningTable();
    }
}